### PR TITLE
Overload DataFrame.drop: sequences must be *str

### PR DESCRIPTION
**Pull request by oliverw1**
The method `DataFrame.drop` expects either 1 `Column`, or 1 `str`, or an iterable of strings. This is only type checked inside the function though. 

Currently the type hints (and the actual API) allow to pass multiple `Column`s but it does result in a runtime error. Personally, I'd like to have that caught earlier. But as this might be getting too close to the internals of the functions, I’d like to hear your opinion on whether or not the type hints should “look inside” to aid development.

### CODE REVIEW

File: test-data/unit/sql-dataframe.test [hunk 1]
```diff
@@ -22,3 +22,18 @@ df.sample(withReplacement=False) # E: No overload variant of "sample" of "DataFr
...
+df.drop("id")
+df.drop("id", "foo")
+df.drop(df.id)
+
+df.drop(col("id"), col("foo"))  # E: No overload variant of "drop" of "DataFrame" matches argument types "Column", "Column" \
+                                # N: Possible overload variant:     \
+                                # N:     def drop(self, *cols: str) -> DataFrame \
+                                # N:     <1 more non-matching overload not shown>
+
+[out]
```
Comment (zero323) [inline test-data/unit/sql-dataframe.test]: Nitpick. Could you add newline here?

**Comment by zero323** [comment]
Nice catch. 

I thought we might resolve it upstream, but this would require an additional signature  in Scala, and it might be a hard to sell.

File: test-data/unit/sql-dataframe.test (hunk 1 above)
Comment (oliverw1) [reply test-data/unit/sql-dataframe.test]: Sure. You have a good eye for detail!

**Comment by oliverw1** [comment]
Well, you carry more weight in the Spark development circles, but I think you're right. It’s a nice to have. With this PR, I’m hoping it eases development for some. Thanks for the review, zero323!

**Comment by zero323** [comment]
> Well, you carry more weight in the Spark development circles, but I think you're right

I seriously doubt that :)

**Comment by zero323** [comment]
Merged into master, backported to 2.4. Thanks your work!

**Pull request merged**